binding-hs/src/Client.hs

```diff
@@ -0,0 +1,192 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Client
```

Comment:
Client makes sense when you are building a local project, but because this is intended to be a library, it might cause naming conflicts. I chose `Network.Gym` because everything here is supported by a flask server (also because I'm thinking of making FFI calls to the gym to bypass the network latency), but `Data.Gym` would work as well. You can also look through some other projects on hackage to see what you think might fit best.